app/screens/settings/notification_auto_responder/notification_auto_responder.tsx

```diff
@@ -67,37 +75,79 @@ const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
-
-    const [notifyProps, setNotifyProps] = useState<Partial<UserNotifyProps>>({
-        ...userNotifyProps,
-        auto_responder_active: (currentUser.status === General.OUT_OF_OFFICE && userNotifyProps.auto_responder_active) ? 'true' : 'false',
-        auto_responder_message: userNotifyProps.auto_responder_message || intl.formatMessage(OOO),
-    });
+    const [autoResponderActive, setAutoResponderActive] = useState((currentUser.status === General.OUT_OF_OFFICE && userNotifyProps.auto_responder_active) ? 'true' : 'false');
+    const [autoResponderMessage, setAutoResponderMessage] = useState(userNotifyProps.auto_responder_message || intl.formatMessage(OOO));
 
     const styles = getStyleSheet(theme);
 
+    const close = () => popTopScreen(componentId);
+
+    const saveButton = useMemo(() => {
+        return {
+            id: SAVE_OOO_BUTTON_ID,
+            enabled: false,
+            showAsAction: 'always' as const,
+            testID: 'notification_settings.auto_res.save.button',
+            color: theme.sidebarHeaderTextColor,
+            text: intl.formatMessage({id: 'settings.save', defaultMessage: 'Save'}),
+        };
+    }, [theme.sidebarHeaderTextColor]);
+
     const onAutoResponseToggle = useCallback((active: boolean) => {
-        setNotifyProps({
-            ...notifyProps,
-            auto_responder_active: `${active}`,
-        });
-    }, [notifyProps]);
+        setAutoResponderActive(`${active}`);
+    }, []);
 
     const onAutoResponseChangeText = useCallback((message: string) => {
-        setNotifyProps({
-            ...notifyProps,
-            auto_responder_message: message,
+        setAutoResponderMessage(message);
+    }, []);
+
+    const saveAutoResponder = useCallback(() => {
+        const notifyProps = {
+            ...userNotifyProps,
+            auto_responder_active: autoResponderActive,
+            auto_responder_message: autoResponderMessage,
+        } as unknown as UserNotifyProps;
+
+        updateMe(serverUrl, {
+            notify_props: notifyProps,
+        });
+        close();
+    }, [serverUrl, autoResponderActive, autoResponderMessage, userNotifyProps]);
+
+    useEffect(() => {
+        const updatedMsg = userNotifyProps?.auto_responder_message !== autoResponderMessage;
+        const enabling = currentUser.status !== General.OUT_OF_OFFICE && autoResponderActive === 'true';
+        const disabling = currentUser.status === General.OUT_OF_OFFICE && autoResponderActive === 'false';
+
+        const enabled = enabling || disabling || updatedMsg;
+
+        const buttons = {
+            rightButtons: [{
+                ...saveButton,
+                enabled,
+            }],
+        };
+        setButtons(componentId, buttons);
+    }, [autoResponderActive, autoResponderMessage, componentId, currentUser.status, userNotifyProps.auto_responder_message]);
+
+    useEffect(() => {
+        setButtons(componentId, {
```

Comment:
I wonder if this effect is really needed given the previous effect

Reply:
I removed it :)